### DISCUSSION TITLE
How to resize multiline text

### DISCUSSION THREAD

**Discussion by birodani3** [pixijs/pixijs#8548] [Q&A]
Hi,

I'm working on a content creation platform where the user can dragndrop frames, images, gifs, texts into the canvas and they can move, resize, rotate them etc. I wrote an algorithm to resize images from any corner of the image, it works perfectly. I use overlay DIV's to control the elements:

![image](https://user-images.githubusercontent.com/17207508/182195299-b204d446-cb5d-4698-bfa9-5c6a4d87ab4c.png)

How could I calculate the fontSize and wordWrapWidth to achieve the desired width and keep the line breaks as they were?

Long story: (my current approach)

- The user grabs one corner and drags it -> I calculate how much width change would be needed (same code as the image resize)
- I use TextMetrics.measureText() to know where the current line breaks are
- I create a new string (lets call it testString) where I place a "\n" to the line break positions.
- I measure the testString's width but I add one to the fontSize and set wordWrap to false so I know how much the width changes per font size change
- From this I can calculate how much I should increase the font size to match the desired width (current width + user drag)

Example:
Current font size: 20px, width: 100px.
User drags the corner and I calculated that the width should change by 20 pixels
I measure my text via TextMetrics.measureText() with fontSize: 20, wordWrap: true, wordWrapWidth 100px to know where are the breaks currently
I measure the "testString" with font size 21, wordWrap:false and it gives me 105px width.
One fontsize-increase results in a 5px width-increase, so to reach 120px I need to increase the fontSize by 4.

The problem:
I dont know how to adjust the wordWrapWidth. In the example I will set fontSize to 24, and I measure my text again and set the wordWrapWidth to Math.ceil(newMetrics.width) but it is not always enough and the texts breaks earlier than it should.

Could somebody help me to find a better approach or to find if I do something wrong? I can provide some code if you want.
Thank you!

**Comment by strong-728** [comment]
hello, i am still facing to that problem now.
if you already fixed that, could you provide the code snipet?
thanks.

**Comment by SignDawn** [comment]
The width of characters may be different
